app/script/telemetry/calling/CallTelemetry.js

```diff
@@ -101,109 +109,68 @@ z.telemetry.calling.CallTelemetry = class CallTelemetry {
-      }
-
-      if ([z.media.MediaType.AUDIO_VIDEO, z.media.MediaType.VIDEO].includes(this.media_type)) {
-        event_name = event_name.replace('_call', '_video_call');
-      }
     }
 
-    amplify.publish(z.event.WebApp.ANALYTICS.EVENT, event_name, attributes);
+    amplify.publish(z.event.WebApp.ANALYTICS.EVENT, eventName, attributes);
   }
 
   /**
    * Track the call duration.
-   * @param {z.calling.entities.CallEntity} call_et - Call entity
+   * @param {z.calling.entities.CallEntity} callEntity - Call entity
    * @returns {undefined} No return value
    */
-  track_duration(call_et) {
-    const {
-      conversationEntity,
-      direction,
-      durationTime,
-      isGroup,
-      terminationReason,
-      timerStart,
-      maxNumberOfParticipants,
-    } = call_et;
+  track_duration(callEntity) {
+    const {terminationReason, timerStart, durationTime} = callEntity;
 
     const duration = Math.floor((Date.now() - timerStart) / 1000);
 
     if (!window.isNaN(duration)) {
       this.logger.info(`Call duration: ${duration} seconds.`, durationTime());
 
-      let duration_bucket;
-      if (duration <= 15) {
-        duration_bucket = '0s-15s';
-      } else if (duration <= 30) {
-        duration_bucket = '16s-30s';
-      } else if (duration <= 60) {
-        duration_bucket = '31s-60s';
-      } else if (duration <= 3 * 60) {
-        duration_bucket = '61s-3min';
-      } else if (duration <= 10 * 60) {
-        duration_bucket = '3min-10min';
-      } else if (duration <= 60 * 60) {
-        duration_bucket = '10min-1h';
-      } else {
-        duration_bucket = '1h-infinite';
-      }
-
       const attributes = {
-        conversation_participants: conversationEntity.getNumberOfParticipants(),
-        conversation_participants_in_call: maxNumberOfParticipants,
-        conversation_type: isGroup
-          ? z.tracking.attribute.ConversationType.GROUP
-          : z.tracking.attribute.ConversationType.ONE_TO_ONE,
-        direction: direction,
-        duration: duration_bucket,
-        duration_sec: duration,
+        AV_switch_toggled: this.hasToggledAV,
+        duration: duration,
         reason: terminationReason,
         remote_version: this.remote_version,
-        with_service: conversationEntity.isWithBot(),
       };
 
-      const isTeamConversation = !!conversationEntity.team_id;
-      if (isTeamConversation) {
-        Object.assign(attributes, z.tracking.helpers.getGuestAttributes(conversationEntity));
-      }
-
-      let event_name = z.tracking.EventName.CALLING.ENDED_CALL;
-      if (this.media_type === z.media.MediaType.AUDIO_VIDEO) {
-        event_name = event_name.replace('_call', '_video_call');
-      }
-
-      amplify.publish(z.event.WebApp.ANALYTICS.EVENT, event_name, attributes);
+      const eventName = z.tracking.EventName.CALLING.ENDED_CALL;
+      this.track_event(eventName, callEntity, attributes);
```

Comment:
I have the feeling, that this fits into a single line.

Reply:
it does indeed